pkg/core/cloudstorage/provider_impl.go

```diff
@@ -81,14 +88,17 @@ func (p *providerImpl) checkDuplicate(assetID string) error {
 	return ErrDuplicateAsset
 }
 
-func (p *providerImpl) Sign(r *SignRequest) (*SignRequest, error) {
+func (p *providerImpl) Sign(scheme string, host string, r *SignRequest) (*SignRequest, error) {
+	now := p.timeProvider.NowUTC()
 	for i, assetItem := range r.Assets {
-		assetID := p.AssetNameToAssetID(assetItem.AssetName)
-		u, err := p.storage.PresignGetObject(assetID)
-		if err != nil {
-			return nil, err
+		u := &url.URL{
+			Scheme: scheme,
+			Host:   host,
+			Path:   fmt.Sprintf("/_asset/get/%s", assetItem.AssetName),
 		}
-		r.Assets[i].URL = u.String()
+		httpRequest, _ := http.NewRequest("GET", u.String(), nil)
+		httpsigning.Sign(p.secret, httpRequest, now, int(PresignGetExpires.Seconds()))
+		r.Assets[i].URL = httpRequest.URL.String()
```

Comment:
The signature of this function seems returns a new sign request, but it actually modify the input request in-place. It would be better to either really return a new request, or remove the request from return value.